System tests: fixes for RHEL8 gating failures

edsantiago opened the pull request:
- tmpfs + noswap test: requires noswap feature in kernel.
  Check for it, and skip if unimplemented. (Root only.
  Rootless test works regardless of kernel).
 
- podman generate systemd tests: always use --files option,
  because otherwise the "DEPRECATED" warning gets written
  to the systemd unit file.
 
- kube play tests: yikes. Fix longstanding bugs when checking
  for containers running. This revealed a longstanding bug
  in one test: multi-pod YAML never actually worked. Fixed now.

- run_podman(): that new check-for-warnings code we added
  in #19878, duh, I skipped it on Debian but should've skipped
  when *runc*. Do so now and update the comment. Requires
  minor surgery to podman_runtime() helper to avoid
  infinite recursion.

Signed-off-by: Ed Santiago <santiago@redhat.com>
```release-note
None
```

mheon commented:
LGTM

Luap99 commented on test/system/060-mount.bats:
```diff
@@ -308,6 +308,11 @@ EOF
 }
 
 @test "podman mount noswap memory mounts" {
+    kern=$(uname -r | awk -F. '{print $1}')
+    if [[ $kern -lt 6 ]]; then
+        skip "tmpfs + noswap requires kernel >= 6.something"
+    fi
```
version checks are bad, especially in a RHEL world where such features may be backported at any time
can you do a check like `mount -t tmpfs -o noswap  none /tmp/....` and if this works we have support for it. (don't forget to umount in this case)

edsantiago replied:
Thanks, I'll try. Am also working on a fix for the multi-pod test, so this is now draft.

Luap99 replied:
Also if you do such a check remember that this is only supported as root, it is not allowed as rootless.

edsantiago commented:
New push:
- functionality test for `noswap` instead of kernel version check
- fixed lots of broken `container wait` loops; I think I found them all but am not sure
- fixed completely-broken-never-actually-worked `kube play` test

Luap99 reviewed (approved):
LGTM
/cherry-pick v4.8

rhatdan commented:
/lgtm

edsantiago commented:
cherrypick is being ignored? It's not a conflict, the commit applies cleanly on my end. I'll submit my own PR.
